lib/auth/clt.go

```diff
@@ -1753,6 +1779,12 @@ type IdentityService interface {
 	// ValidateSAMLResponse validates SAML auth response
 	ValidateSAMLResponse(re string) (*SAMLAuthResponse, error)
 
+	// GetSAMLAuthRequest returns SAML auth request if found
+	GetSAMLAuthRequest(id string) (*services.SAMLAuthRequest, error)
```

Comment:
The context should really be propagated here by the caller. This doesn't look like it is used frequently enough that adding this here and to the `IdentityService` will cause this PR to get too noisy.
```suggestion
	GetSAMLAuthRequest(ctx context.Context, id string) (*services.SAMLAuthRequest, error)
```

Reply:
Done for this and all methods calling `GetSAMLAuthRequest`.